pyproject.toml

```diff
@@ -265,8 +265,6 @@ ignore = [
     "UP006", # use-pep585-annotation
     "UP007", # use-pep604-annotation
     # https://beta.ruff.rs/docs/rules/#flake8-type-checking-tch
-    # doesn't catch errors or help with circular imports and tedious to apply
-    "TCH002", # typing-only-third-party-import
```

Comment:
As of ruff `0.0.271` auto-fix rules exist for this rule.
So turning this on.
It can prevent missing dependency issues if the dependency was just used as an annotation.

We've experienced this problem in particular with `typing_extensions` which is a common cause of failures on older versions of Airflow.